pkgs/development/python-modules/pylibftdi/default.nix

```diff
@@ -0,0 +1,52 @@
+{ stdenv
+, buildPythonPackage
+, fetchPypi
+, pythonAtLeast
+, httpserver
+, libftdi1
+, libusb1
+}:
+
+buildPythonPackage rec {
+  pname = "pylibftdi";
+  version = "0.17.0";
+  pythonAtLeast = "3"; # compatibility with httpserver
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "132c44rs0f2zm22hnhg93p47aaw6llpfzavc64dcqrn8719mlqfw";
+  };
+
+  propagatedBuildInputs = [
+    httpserver
+    libftdi1
+    libusb1
+  ];
+
+  # Several modules have been renamed for python3. ReadTimeout doesn't
+  # appear to be part of pylibftdi at all...
+  postPatch = ''
+    substituteInPlace pylibftdi/examples/bit_server.py --replace \
+      "BaseHTTPServer" "http.server"
+
+    substituteInPlace pylibftdi/examples/bit_server.py --replace \
+      "cStringIO" "io"
+
+    substituteInPlace pylibftdi/examples/bit_server.py --replace \
+      "SocketServer" "socketserver"
+
+    substituteInPlace pylibftdi/examples/serial_transfer.py --replace \
+      ", ReadTimeout" ""
+
+    substituteInPlace pylibftdi/driver.py --replace \
+      "find_library(dll)" \
+      "'${libftdi1.out}/lib/libftdi1.so' if dll in self._lib_search['libftdi'] else '${libusb1.out}/lib/libusb-1.0.so' if dll in self._lib_search['libusb'] else find_library(dll)"
+  '';
```

Comment:
I'm not the biggest fan of the nested if, I think it would be slightly better to create a patch and just have 

```
  patches = [
    ./make-python3-compatible.patch
    ./fix-search-paths.patch
  ];
```